src/lib.rs

```diff
@@ -743,7 +750,35 @@ fn starts_with_plus_eq(text: &OsStr, prefix: &str) -> bool {
     false
 }
 
-#[cfg(feature = "eq-separator")]
+#[cfg(feature = "short-space-opt")]
+#[inline(never)]
+fn starts_with_short_prefix(text: &OsStr, prefix: &str) -> bool {
+    if prefix.starts_with("--") {
+        return false; // Only works for short keys
+    }
+    if let Some(s) = text.to_str() {
+        if s.get(0..prefix.len()) == Some(prefix) {
+            return true;
+        }
+    }
+
+    false
+}
+
+#[cfg(all(feature = "eq-separator", feature = "short-space-opt"))]
+fn index_predicate(text: &OsStr, prefix: &str) -> bool {
```

Comment:
All three of this methods can be inlined.

Reply:
That's what I thought, but I wasn't sure if inlining worked in Rust the same as it does in C.